### ISSUE TITLE
Transaction commit /  rollback methods are missing proper documentation

### ISSUE THREAD

**Issue by elhigu** [knex/knex#5822]
At least required error parameter of rollback should be documented. Maybe add also disclaimer that 2 different transaction styles should not be mixed, or one will get strange results (there could be check for it in knex too if promise is returned from transaction, then explicit rollback/commit would fail).

**Comment by mercmobily** [comment]
You're absolutely right that the transaction documentation is missing important details about proper usage of therollback and commit methods.

This issue has been addressed in the current documentation at docs/src/guide/transactions.md:139, which now states: "Calling `trx.rollback` will return a rejected Promise. If you don't pass any argument to `trx.rollback`, a generic `Error` object will be created and passed in to ensure the Promise always rejects with something."

Looking at the implementation in lib/execution/transaction.js:142-151, the rollback method indeed requires an error parameter for proper operation. When rollback is called without an error, the code at line 185 creates a generic error: `new Error('Transaction rejected with non-error: ${value}')`. This behaviour ensures the transaction promise properly rejects.

Regarding the mixing of transaction styles, the code at lib/execution/transaction.js:237-244 shows that when a promise is returned from the transaction handler, Knex automatically chains commit and rollback to the promise's resolution and rejection. This means if you return a promise from your transaction handler, you should not manually call commit or rollback, as they will be called automatically based on the promise's outcome.

The test suite demonstrates proper usage patterns. At test/tape/transactions.js, we can see explicit rollback being used with an error parameter: `trx.rollback(new Error('Rolled back'))`. This test specifically validates that transactions handle explicit rollback calls correctly and skip any subsequent operations after rollback is called.

For the two transaction styles mentioned:

Style 1 (Automatic): Return a promise from the transaction handler, and Knex handles commit/rollback automatically based on promise resolution/rejection.

Style 2 (Manual): Don't return a promise from the handler, manually chain .then(trx.commit).catch(trx.rollback) to your query promises.

The critical point is that these styles should not be mixed within the same transaction. If you return a promise from the transaction handler (Style 1), Knex will automatically manage the transaction lifecycle, and manually calling commit or rollback could lead to unexpected behavior or errors.

The documentation has been updated to clarify these points, though it could potentially be enhanced further with explicit warnings about not mixing transaction styles and clearer examples of the error parameter requirement for rollback.

Related Issues:
- #785 - Addressed skipping extra transaction statements after commit/rollback
- Various transaction handling improvements in recent versions
